src/Pos/Generator/BlockEvent.hs

```diff
@@ -168,8 +174,7 @@ getBlockIndexRange ::
        [BlockEvent' BlockIndex]
     -> (BlockIndex, BlockIndex)
 getBlockIndexRange =
-    let minMax a = Smg.Option (Just (Smg.Min a, Smg.Max a))
-    in maybe (0, 0) (over _2 (+1)) . coerce . foldMap minMax . Compose
+    maybe (0, 0) (\(lower, upper) -> (lower, upper+1)) . minMaxOf (folded.folded)
```

Comment:
Yes, ha ha. I asked @neongreen how he would write it and he recommended lenses (unsurprisingly).